src/zkevm_specs/evm_circuit/execution/memory.py

```diff
@@ -22,18 +30,32 @@ def memory(instruction: Instruction):
     )
 
     if is_mstore8 == FQ(1):
-        instruction.is_equal(
-            instruction.memory_lookup(RW.Write, address.expr()), FQ(value.le_bytes[0])
-        )
+        value_left = instruction.memory_lookup(RW.Write, addr_left)
+        # TODO: check with mask
 
     if is_not_mstore8 == FQ(1):
-        for idx in range(32):
-            instruction.is_equal(
-                instruction.memory_lookup(
-                    RW.Write if is_store == FQ(1) else RW.Read, address.expr() + idx
-                ),
-                FQ(value.le_bytes[31 - idx]),
-            )
+        value_left = instruction.memory_lookup(
+            RW.Write if is_store == FQ(1) else RW.Read, addr_left
+        )
+        value_right = instruction.memory_lookup(
+            RW.Write if is_store == FQ(1) else RW.Read, addr_right
+        )
+
+        # Check consistency of value, value_left, and value_right.
+        mask = make_mask(shift)
+        X = instruction.randomness
```

Comment:
memory opcodes(mload/mstore/mstore8) don't use copy circuit. I conclude if only refactor these memory opcodes, just evm execution circuit and state circuit require refactor accordingly. copy circuit doesn't need to change.  only when update execution codes(like calldatacopy, log* etc.) requires copy circuit udpates.

Reply:
The copy circuit needs to change because the memory format changed. It must read and write the same way that the memory instructions do, so by word instead of by byte.

This is not included in this PR yet.